### PR TITLE
Folders

### LINKED ISSUE #80: Add folders to the file system

### PR DESCRIPTION
**Pull request by curran**
image

image

Closes #80 

Summary of changes:

 * Created tests for the following cases and made them pass:
   * Empty directory case
   * Full directory case
   * Recursive case
 * For now, directories are explicitly represented in the data structure as "files" where `file.text = null` and `file.name` has a trailing slash.
 * In the future we may want to drop the explicit representation, but maybe it's fine to keep it? Not sure. In any case, from the perspective of reading from the file system, the simpler option is to keep it. We can always filter it out in the sidebar logic if it's problematic.
 * Added arrows and styling
 * Implement opening and closing of folders

Follow-on tasks:
 * Ensure text selection doesn't happen when you open and close folders #98
 * If we eventually add a stateful URL that contains which file is open, initialize the open folders to have the folders that contain that file automatically open on page load #99
 * Delete directory #102
 * Rename directory #103
 * Create directory #104
 * Drag & drop files between directories #105
 * Drag & drop directories between directories #106

### CODE REVIEW

**Comment by Anooj-Pai** [comment]
Screenshot 2023-04-25 at 8 54 52 AM

I get this error when I try and run it

**Comment by curran** [comment]
Interesting. That seems unrelated to these changes. Did you run `npm install`? Does it happen on `main` also?

**Comment by curran** [comment]
@Anooj-Pai This is ready for re-review. Thanks!

For that strange error, I'm not sure what could be the cause. Maybe try deleting `node_modules` and running `npm install`? Also can you tell me what your `node --version` is?

**Comment by curran** [comment]
OK the core of this is working!

image

Remaining tasks for this PR:
 * [ ] Sort the file tree
 * [ ] Implement opening and closing of directories (see VizHub2 codebase)
 * [ ] Fix layout issue with utils buttons
 * [ ] Add proper icons for open/closed directories (see VizHub2 codebase)

Tasks for future issues:
 * [ ] Create directory interaction
 * [ ] Rename directory interaction
 * [ ] Delete directory interaction
 * [ ] Drag & drop files between directories
 * [ ] Drag & drop directories between directories

File: src/client/App.jsx [hunk 1]
```diff
@@ -208,6 +208,11 @@ function App() {
     setSettings(false);
   }, []);
 
+  const fileNameSplit = (fileName) => {
```
Comment (curran) [inline src/client/App.jsx:211]: Very nice! Thank you for this.

**Comment by curran** [comment]
@Anooj-Pai This is ready for review. Folders are working woohoo!

**Review by Anooj-Pai — APPROVED** [review]
This looks great! amazing work and this honestly puts us in a place where it is fully functional for the public. Only a few things left to do!

**Pull request merged**